### ISSUE TITLE
Avoid capy/bara clashes with actions on forks

### ISSUE THREAD

**Issue by wdconinc** [eic/EICrecon#867]
**Is your feature request related to a problem? Please describe.**
The pipeline at https://github.com/eic/EICrecon/actions/runs/5956311776/job/16157787444?pr=840#step:8:23 pulled in the latest artifacts from target branch main, except it picked the main branch of @ruse-traveler since that CI pipeline was more recent than the most recent eic main pipeline...

**Describe the solution you'd like**
Work with https://github.com/dawidd6/action-download-artifact to introduce a flag that blocks forks from being considered, or work with @ruse-traveler to avoid using main branches on forks (maybe enforced through CI checks; forks would still be allowed to use other branches, just not main due to these clashes).

**Describe alternatives you've considered**
See above.

**Comment by wdconinc** [comment]
FYI @veprbl. Know any typescript for a PR to https://github.com/dawidd6/action-download-artifact/blob/master/main.js? ;-)

**Comment by wdconinc** [comment]
See also https://github.com/eic/EICrecon/pull/840#issuecomment-1675291500, which was already leading to confusion.

**Comment by ruse-traveler** [comment]
I apologize. 😓 

I'll avoid using my main for PRs from now on...

**Comment by veprbl** [comment]
capy rewrites fetching of artifacts in python, but, unfortunately, has the same defect. That depends on https://github.com/eic/epic-capybara/issues/2. I assume, merge-base can't be calculated by querying GitHub API. For sure, we can fetch the repo and run git on it, but I haven't had time to implement that.

**Comment by veprbl** [comment]
Frankly, the only way to ensure comparable artifacts is to always run tests for the merge-base in parallel.

**Comment by wdconinc** [comment]
We're already squeezed for ci minutes...

**Comment by veprbl** [comment]
> We're already squeezed for ci minutes...

Part of the squeezing happens because of the need to re-run workflows because comparisons don't match or just suspicious. We could also cut down on gcc vs clang runs.

**Comment by wdconinc** [comment]
Getting a bit out of scope for this issue, but I think there's benefit in gcc v clang. If cutting, I'd look at the multiple DIS sets. But ideally, I wouldn't want to cut at all and expand our capacity... Confirmed on Monday that this would be within OSG ToS/acceptable use policy if we implemented a scheme like we talked about.
